src/datatypes/epgsql_codec_timerange.erl

```diff
@@ -0,0 +1,51 @@
+%%% @doc
+%%% Codec for `tsrange', `tstzrange', `daterange' types.
+%%% https://www.postgresql.org/docs/current/static/rangetypes.html#rangetypes-builtin
+%%% $PG$/src/backend/utils/adt/rangetypes.c
+%%% @end
+%%% Created : 16 Jul 2018 by Vladimir Sekissov <eryx67@gmail.com>
+%%% TODO: universal range, based on pg_range table
+%%% TODO: inclusive/exclusive ranges `[]' `[)' `(]' `()'
+
+-module(epgsql_codec_timerange).
+-behaviour(epgsql_codec).
+
+-export([init/2, names/0, encode/3, decode/3, decode_text/3]).
+
+-include("protocol.hrl").
+
+-export_type([data/0]).
+
+-type data() :: {epgsql_codec_datetime:data(), epgsql_codec_datetime:data()} | empty.
+
+init(_, Sock) ->
+    case epgsql_sock:get_parameter_internal(<<"integer_datetimes">>, Sock) of
+        <<"on">>  -> epgsql_idatetime;
+        <<"off">> -> epgsql_fdatetime
+    end.
+
+names() ->
+    [tsrange, tstzrange, daterange].
+
+encode(empty, _T, _CM) ->
+    <<1>>;
+encode({From, To}, Type, CoderMod) ->
+    FromBin = CoderMod:encode(member_type(Type), From),
+    ToBin = CoderMod:encode(member_type(Type), To),
```

Comment:
I expect elvis to complain about dynamic calls. If it will, you can try to rewrite this to call `epgsql_codec_datetime` as `epgsql_codec_datetime:encode(To / From, member_type(Type), CoderMod)` or maybe even do smth like:

```
init(_Args, Sock) ->
    epgsql_codec_datetime:init(_Args, Sock).

encode({From, To}, Type, DtCodec) ->
    FromBin = epgsql_codec:encode(epgsql_codec_datetime, From, member_type(Type), DtCodec),
    ToBin = ...
    <..>.
decode(..., DtCodec) ->
    From = epgsql_codec:decode(epgsql_codec_datetime, FromBin, member_type(Type), DtCodec),
    To = ...
    <..>.
```

Or just create helper functions that will have clauses for integer or float callbacks and will call them directly.